### PR TITLE
Fix #5393: Make hasProtoExtra centralized.

### LINKED ISSUE #5393: [Feature Request]:  Make hasProtoExtra centralized.
### Is your feature request related to a problem? Please describe.

Move `hasProtoExtra` to central utility since it will be widely used once we shift all intent extras and arguments over protos.

### Describe the solution you'd like

N/A

### Describe alternatives you've considered

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Vishwajith-Shettigar**
## Explanation
Fixes #5393
Created `ProtoExtraMatcher` class to make `hasProtoExtra` matcher centralized.

## Essential Checklist

- [ ] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [ ] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [ ] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by Vishwajith-Shettigar** [comment]
@adhiamboperes, PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/utility/ProtoExtraMatcher.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import com.google.protobuf.MessageLite
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
+import org.oppia.android.util.extensions.getProtoExtra
+
+/** A custom matcher to check if an Intent has a specific proto extra. */
+class ProtoExtraMatcher {
+  companion object {
+    fun <T : MessageLite> hasProtoExtraCheck(keyName: String, expectedProto: T):
```
Comment (adhiamboperes) [inline app/src/sharedTest/java/org/oppia/android/app/utility/ProtoExtraMatcher.kt]: Please add a kdoc to this function and remove the kdoc exemption.

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @Vishwajith-Shettigar! This is a  really nice optimization for the developers now that we are moving to protos in intents.

I just had one follow up.

File: app/src/sharedTest/java/org/oppia/android/app/utility/ProtoExtraMatcher.kt (hunk 1 above)
Comment (Vishwajith-Shettigar) [reply app/src/sharedTest/java/org/oppia/android/app/utility/ProtoExtraMatcher.kt]: Done.

Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/utility/ProtoExtraMatcher.kt]: @Vishwajith-Shettigar, you seem to have missed removing the file from exemptions. See static check failure.

Comment (Vishwajith-Shettigar) [reply app/src/sharedTest/java/org/oppia/android/app/utility/ProtoExtraMatcher.kt]: ahh!, right. I'll remove, thank you.

**Review by adhiamboperes — APPROVED** [review]
Thanks @Vishwajith-Shettigar! LGTM

**Pull request merged**